Vector*: Added random()

EthanHermsey opened the pull request:
Fixed #18996.

Adding a method that sets the components of the vector with a random
number [0-1].

EthanHermsey committed:
added random method to vectors

Adding a method that sets the components of the vector with a random
number [0-1]

Mugen87 commented on src/math/Vector2.d.ts:
```diff
@@ -457,6 +457,11 @@ export class Vector2 implements Vector {
 	 * @see {@link http://en.wikipedia.org/wiki/Taxicab_geometry|Wikipedia: Taxicab Geometry}
 	 */
-	manhattanDistanceTo( v: Vector2 ): number;
+    manhattanDistanceTo( v: Vector2 ): number;
```
Please use hard instead of soft tabs.

Mugen87 commented:
It would be great if you can also update the docs 👍 


https://threejs.org/docs/index.html#api/en/math/Vector2
https://threejs.org/docs/index.html#api/zh/math/Vector2 (just copy over the English version)
https://threejs.org/docs/index.html#api/en/math/Vector3
https://threejs.org/docs/index.html#api/zh/math/Vector3
https://threejs.org/docs/index.html#api/en/math/Vector4
https://threejs.org/docs/index.html#api/zh/math/Vector4

EthanHermsey committed:
Hard tabs and updated docs.

EthanHermsey commented:
I have added the hard tabs and changed the docs. Is it done correctly?
I am not certain about the like 'Don't include build files in your Pull Request' on your article about contributing. Those files are forked, do it remove them or do you mean don't change anything inside the build files? 

Also, in the documentation of all vectors, the use of 'the components of THIS vector' and 'the components of THE vector' are used interchangeably. Just wanted to point that out.

Mugen87 commented:
> Those files are forked, do it remove them or do you mean don't change anything inside the build files?

The idea is not to include files from the `build` directory in PRs. You are not doing this so the PR looks fine.

> Also, in the documentation of all vectors, the use of 'the components of THIS vector' and 'the components of THE vector' are used interchangeably. Maybe it's nice to use only one of them?

Yes, improve the wording is a good idea. But please in a separate PR.

Mugen87 commented on src/math/Vector2.js:
```diff
@@ -482,6 +482,15 @@ Object.assign( Vector2.prototype, {
 		return this;
 
+	},
+	
```
Please remove all trailing spaces from your PR.

BTW: If you work with editors like VSCode, this is done automatically by the eslint autocorrect.

EthanHermsey replied:
Hhmm I tried this with the default eslint settings. It does remove the trails but messes up the rest of the document. For now I did it by hand, I will have to figure this out, using the eslint-config-mdcs. It's very nice.

EthanHermsey committed:
Removed trailing spaces.

mrdoob commented:
Thanks!
